tests/mock/test_to_pandas.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+import datetime
+
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from snowflake.snowpark import Session
+from snowflake.snowpark.mock.connection import MockServerConnection
+
+session = Session(MockServerConnection())
+
+
+@pytest.mark.localtest
+def test_df_to_pandas_df():
+    df = session.create_dataframe(
```

Comment:
Can we cover more types, i.e. https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/types except for Geography and Geometry?

Reply:
updated